### PR TITLE
[wip] codecov

### PR DESCRIPTION
**Pull request by dpgraham**
Adds coverage reports to pull requests

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -0,0 +1,16 @@
+assets/
```
Comment (imurchie) [inline .npmignore]: This file is ignored. Having it will be confusing.

File: .travis.yml [hunk 2]
```diff
@@ -13,7 +13,10 @@ jobs:
   include:
     - stage: Tests
       os: linux
-      script: npm run test && npm run e2e-test
+      script: |
+        npm run e2e-test
+        npx gulp coverage
+        npx codecov
       node_js: "11"
       after_script: npm run coverage
```
Comment (imurchie) [inline .travis.yml]: Are we going to get rid of coveralls, then?

**Review by imurchie — COMMENTED** [review]
I'd rather this be moved into `appium-gulp-plugins` so the coverage deps and whatnot are all kept in one place, and usable throughout.

**Comment by dpgraham** [comment]
@imurchie yeah I just wanted to try it out. Maybe coveralls provides code coverage analysis within pull requests?

**Comment by imurchie** [comment]
I feel like it used to, but I could be mistaken. If there is a better solution, I'm all for it, though.

Experiment away! Once it works well we can move it over (and put it under what we already map to `npm run coverage`, so we don't need to do anything in the packages that already do that.

**Comment by dpgraham** [comment]
I just had to enable it on Coveralls (see example here: https://github.com/appium/appium/pull/12494#issuecomment-482669164)

Closing this in favour of keeping Coveralls.

**Pull request closed without merging**